### PR TITLE
Add new test case host_verify_saml_token

### PR DESCRIPTION
**Pull request by keirazhang**
Add a new test case to test host verified SAML token when performing VGAuth guest operations.

Tested on Windows:
```
+------------------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                                           |
|                           | bitness='64'                                                 |
|                           | buildNumber='26100'                                          |
|                           | distroName='Windows'                                         |
|                           | distroVersion='10.0'                                         |
|                           | familyName='Windows'                                         |
|                           | kernelVersion='26100.560'                                    |
|                           | prettyName='Windows 11 Enterprise, 64-bit (Build 26100.560)' |
+------------------------------------------------------------------------------------------+


Test Results (Total: 29, Passed: 28, Failed: 1, Elapsed Time: 03:37:35)
+--------------------------------------------------------------+
| ID | Name                             |   Status | Exec Time |
+--------------------------------------------------------------+
| 01 | deploy_vm_efi_nvme_e1000e        |   Passed | 00:18:25  |
| 02 | check_inbox_driver               |   Passed | 00:07:49  |
| 03 | secureboot_enable_disable        |   Passed | 00:11:11  |
| 04 | wintools_complete_install_verify |   Passed | 00:10:51  |
| 05 | check_efi_firmware               |   Passed | 00:01:02  |
| 06 | check_ip_address                 |   Passed | 00:01:07  |
| 07 | check_os_fullname                |   Passed | 00:00:47  |
| 08 | mouse_driver_vmtools             |   Passed | 00:00:43  |
| 09 | vgauth_check_service             |   Passed | 00:00:48  |
| 10 | host_verify_saml_token           |   Passed | 00:05:48  |
| 11 | stat_balloon                     |   Passed | 00:00:43  |
| 12 | stat_hosttime                    |   Passed | 00:01:05  |
| 13 | paravirtual_vhba_device_ops      |   Passed | 00:09:17  |
| 14 | lsilogicsas_vhba_device_ops      |   Passed | 00:08:57  |
| 15 | sata_vhba_device_ops             |   Passed | 00:09:22  |
| 16 | nvme_vhba_device_ops             |   Passed | 00:09:24  |
| 17 | nvme_vhba_device_ops_spec13      |   Passed | 00:12:41  |
| 18 | nvme_disk_hot_extend_spec13      |   Passed | 00:08:36  |
| 19 | nvdimm_cold_add_remove           | * Failed | 00:05:06  |
| 20 | e1000e_network_device_ops        |   Passed | 00:07:17  |
| 21 | vmxnet3_network_device_ops       |   Passed | 00:07:33  |
| 22 | memory_hot_add_basic             |   Passed | 00:07:13  |
| 23 | check_quiesce_snapshot           |   Passed | 00:05:06  |
| 24 | cpu_multicores_per_socket        |   Passed | 00:16:07  |
| 25 | gosc_sanity_staticip             |   Passed | 00:08:11  |
| 26 | gosc_sanity_dhcp                 |   Passed | 00:16:42  |
| 27 | vtpm_cold_add_remove             |   Passed | 00:09:07  |
| 28 | wintools_uninstall_verify        |   Passed | 00:10:20  |
| 29 | cpu_hot_add_basic                |   Passed | 00:04:48  |
+--------------------------------------------------------------+
```

Tested on RHEL 9
```
+----------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                                   |
|                           | bitness='64'                                         |
|                           | cpeString='cpe:/o:redhat:enterprise_linux:9::baseos' |
|                           | distroAddlVersion='9.4 (Plow)'                       |
|                           | distroName='Red Hat Enterprise Linux'                |
|                           | distroVersion='9.4'                                  |
|                           | familyName='Linux'                                   |
|                           | kernelVersion='5.14.0-427.13.1.el9_4.x86_64'         |
|                           | prettyName='Red Hat Enterprise Linux 9.4 (Plow)'     |
+----------------------------------------------------------------------------------+


Test Results (Total: 29, Passed: 27, Skipped: 2, Elapsed Time: 01:47:49)
+-------------------------------------------------------------------------+
| ID | Name                                 |   Status        | Exec Time |
+-------------------------------------------------------------------------+
| 01 | ovt_verify_pkg_install               |   Passed        | 00:06:24  |
| 02 | ovt_verify_status                    |   Passed        | 00:02:37  |
| 03 | vgauth_check_service                 |   Passed        | 00:00:51  |
| 04 | host_verify_saml_token               |   Passed        | 00:01:40  |
| 05 | check_ip_address                     |   Passed        | 00:00:40  |
| 06 | check_os_fullname                    |   Passed        | 00:01:04  |
| 07 | stat_balloon                         |   Passed        | 00:00:42  |
| 08 | stat_hosttime                        |   Passed        | 00:00:42  |
| 09 | device_list                          |   Passed        | 00:02:22  |
| 10 | check_quiesce_snapshot_custom_script |   Passed        | 00:01:06  |
| 11 | memory_hot_add_basic                 |   Passed        | 00:05:09  |
| 12 | cpu_hot_add_basic                    |   Passed        | 00:04:44  |
| 13 | cpu_multicores_per_socket            |   Passed        | 00:07:17  |
| 14 | check_efi_firmware                   |   Passed        | 00:00:47  |
| 15 | secureboot_enable_disable            |   Passed        | 00:03:55  |
| 16 | e1000e_network_device_ops            |   Passed        | 00:03:53  |
| 17 | vmxnet3_network_device_ops           |   Passed        | 00:02:51  |
| 18 | pvrdma_network_device_ops            |   Passed        | 00:07:01  |
| 19 | gosc_perl_dhcp                       |   Passed        | 00:05:34  |
| 20 | gosc_perl_staticip                   |   Passed        | 00:05:10  |
| 21 | gosc_cloudinit_dhcp                  |   Passed        | 00:05:56  |
| 22 | gosc_cloudinit_staticip              |   Passed        | 00:06:14  |
| 23 | paravirtual_vhba_device_ops          |   Passed        | 00:06:09  |
| 24 | lsilogic_vhba_device_ops             | * Not Supported | 00:00:37  |
| 25 | lsilogicsas_vhba_device_ops          | * Not Supported | 00:00:39  |
| 26 | sata_vhba_device_ops                 |   Passed        | 00:06:38  |
| 27 | nvme_vhba_device_ops                 |   Passed        | 00:06:18  |
| 28 | nvdimm_cold_add_remove               |   Passed        | 00:06:48  |
| 29 | ovt_verify_pkg_uninstall             |   Passed        | 00:02:48  |
+-------------------------------------------------------------------------+
```

### CODE REVIEW

File: common/vm_check_vmtools_capability.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+- name: "Check VMware Tools capability {{ vmtools_capability_key }} is TRUE"
+  ansible.builtin.assert:
+    that:
+      - vmtools_capability_key in vm_extra_config
+      - vm_extra_config[vmtools_capability_key] == "TRUE"
+    fail_msg: >-
+      VMware Tools capability '{{ vmtools_capability_key }} = TRUE' doesn't exsit in VM's advance settings.
+      Current {{ vmtools_capability_key }} is {{ vm_extra_config[vmtools_capability_key] | default('undefined') }}
+    success_msg: >-
+      VMware Tools capability '{{ vmtools_capability_key }}' is TRUE in VM's advance settings
```
Comment (Tomorrow9) [inline common/vm_check_vmtools_capability.yml:24]: advanced

File: linux/host_verify_saml_token/host_verify_saml_token.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    - name: "Test case block"
+      block:
+        - name: "Skip test case due to missing vCenter Server variables"
+          include_tasks: ../../common/skip_test_case.yml
+          vars:
+            skip_msg: "Skip test case {{ ansible_play_name }} is because of missing vCenter Server variables."
+            skip_reason: "Not Applicable"
+          when: >
+            (vcenter_is_defined is undefined or not vcenter_is_defined) or
+            (vcenter_ssh_username is undefined or not vcenter_ssh_username) or
```
Comment (Tomorrow9) [inline linux/host_verify_saml_token/host_verify_saml_token.yml:20]: Where are these 2 parameters 'vcenter_ssh_username' and 'vcenter_ssh_password' been used?

Comment (keirazhang) [reply linux/host_verify_saml_token/host_verify_saml_token.yml:20]: `common/add_vsphere_hosts_in_inventory.yml` added vcenter_hostname to inventory, and `common/vcenter_manage_domain_user.yml` will use it for adding VC domain user

File: common/vm_check_vmtools_capability.yml (hunk 1 above)
Comment (keirazhang) [reply common/vm_check_vmtools_capability.yml]: Updated. Thanks!

**Review by Tomorrow9 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**